### ISSUE TITLE
Support in-place updates of container resource constraints

### ISSUE THREAD

**Issue by mhausenblas** [kubernetes-sigs/cri-o#719]
It would be very beneficial if CRI-O could solve https://github.com/kubernetes/kubernetes/issues/5774, that is, rather than having to launch a new pod when changing the resource limits, modifying the CPU and memory constraints of a running container.

CC: @rhatdan

**Comment by rhatdan** [comment]
@mrunalp @runcom WDYT?

**Comment by runcom** [comment]
I +1'ed this already, I'm super in favor of something like - only thing I hadn't understood if we're blocked on upstream kubernetes (because of the scheduler) or on container runtimes (cri-o and others)

**Comment by rhatdan** [comment]
If there is an API to do this, we should, if not we should at least implement it in kpod.

**Comment by mrunalp** [comment]
runc has an update functionality. With k8s, scheduling and other decisions come into play. It should be straight forward for cri-o to implement it once we get the call over CRI, but kubelet and higher layers would have to make the decisions.

**Comment by mhausenblas** [comment]
So thanks folks, great to learn that CRI-O in principle can address this. Since it seems it's more a question what the dependencies are (like: what needs to change in the kubelet) may I suggest that we comment on https://github.com/kubernetes/kubernetes/issues/5774 to clarify this?

**Comment by mhausenblas** [comment]
Any updates here?

**Comment by mhausenblas** [comment]
@rhatdan do we have a timeline for this?

**Comment by rhatdan** [comment]
@mhausenblas I don't really work at that level.  Probably @smarterclayton @derekwaynecarr @eparis would be better to move it forward.

**Comment by mhausenblas** [comment]
Bumping this again, we should get our act together here and at least have a timeline for it. See also https://github.com/kubernetes/community/pull/1719

**Comment by rhatdan** [comment]
@mhausenblas CRI-O just does stuff that is implemented in the CRI by Kubernetes, if Kubernetes grows the ability to modify the resource constraints in CRI, the CRI-O Will implement them.

**Comment by rhatdan** [comment]
No action in another year.  So closing this as won't fix until Kubernetes requires.  Reopen if you disagree.

**Comment by johanneswuerbach** [comment]
@rhatdan in-place resource changes are making progress in kubernetes https://github.com/kubernetes/enhancements/pull/686 so this will hopefully be required soon.
